examples/apps/calendar-explorer/src/main.rs

```diff
@@ -0,0 +1,241 @@
+        match self {
+            Self::Default => Self::Surrounding,
+            Self::Surrounding => Self::WeekdaysHeader,
+            Self::WeekdaysHeader => Self::SurroundingAndWeekdaysHeader,
+            Self::SurroundingAndWeekdaysHeader => Self::MonthHeader,
+            Self::MonthHeader => Self::MonthAndWeekdaysHeader,
+            Self::MonthAndWeekdaysHeader => Self::Default,
+        }
+    }
+}
+
+impl fmt::Display for StyledCalendar {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Default => write!(f, "Default"),
+            Self::Surrounding => write!(f, "Show Surrounding"),
+            Self::WeekdaysHeader => write!(f, "Show Weekdays Header"),
+            Self::SurroundingAndWeekdaysHeader => write!(f, "Show Surrounding and Weekdays Header"),
+            Self::MonthHeader => write!(f, "Show Month Header"),
+            Self::MonthAndWeekdaysHeader => write!(f, "Show Month Header and Weekdays Header"),
+        }
+    }
+}
+
+impl StyledCalendar {
+    fn render_year(&self, frame: &mut Frame, area: Rect, date: Date) -> Result<()> {
+        let events = events(date)?;
+
+        let area = area.inner(Margin {
+            vertical: 1,
+            horizontal: 1,
+        });
+        let rows = Layout::vertical([Constraint::Ratio(1, 3); 3]).split(area);
+        let areas = rows.iter().flat_map(|row| {
+            Layout::horizontal([Constraint::Ratio(1, 4); 4])
+                .split(*row)
+                .to_vec()
+        });
+        for (i, area) in areas.enumerate() {
+            let month = date
+                .replace_day(1)
+                .unwrap()
+                .replace_month(Month::try_from(i as u8 + 1).unwrap())
+                .unwrap();
+            self.render_month(frame, area, month, &events);
+        }
+        Ok(())
+    }
+
+    fn render_month(&self, frame: &mut Frame, area: Rect, date: Date, events: &CalendarEventStore) {
+        let calendar = Monthly::new(date, events)
+            .default_style(Style::new().bold().bg(Color::Rgb(50, 50, 50)))
+            .show_month_header(Style::default());
+        let calendar = match self {
+            StyledCalendar::Default => calendar,
+            StyledCalendar::Surrounding => calendar.show_surrounding(Style::new().dim()),
+            StyledCalendar::WeekdaysHeader => {
+                calendar.show_weekdays_header(Style::new().bold().green())
+            }
+            StyledCalendar::SurroundingAndWeekdaysHeader => calendar
+                .show_surrounding(Style::new().dim())
+                .show_weekdays_header(Style::new().bold().green()),
+            StyledCalendar::MonthHeader => calendar.show_month_header(Style::new().bold().green()),
+            StyledCalendar::MonthAndWeekdaysHeader => {
+                calendar.show_weekdays_header(Style::new().bold().dim().light_yellow())
+            }
+        };
+        frame.render_widget(calendar, area);
+    }
+}
```

Comment:
There were a bit of code repetition but my reasoning for having separate functions was making them easy to copy/paste.

Consider this scenario:

1. User runs the calendar explorer.
2. Likes one of the styles and wants to incorporate that in their app.
3. Simply copies the function to their app.

I guess you could argue that they should just _learn the API_ ™️ to make proper use of the widgets. But most people are lazy and just want to build something quickly.

Also, this is an "app example" so it should be fine. It is still an example though...

What do you think?

Reply:
For the explorer apps, I'm less concerned about copy paste, but perhaps inlining the first calendar line might be sufficient here. Mostly I didn't like that it was random methods labelled example1, example2, ...

https://github.com/ratatui/ratatui/pull/1571/commits/315393e2e12e6c4fe056be14664168daf5b519eb